staging/src/k8s.io/client-go/metadata/metadatainformer/interface.go

```diff
@@ -27,6 +27,7 @@ type SharedInformerFactory interface {
 	Start(stopCh <-chan struct{})
 	ForResource(gvr schema.GroupVersionResource) informers.GenericInformer
 	WaitForCacheSync(stopCh <-chan struct{}) map[schema.GroupVersionResource]bool
+	Shutdown()
```

Comment:
naive by-stander question, why do we need a shutdown when starts already takes a `stopCh`?

Reply:
The shutdown additionally actually waits for everything to terminate is the difference

Reply:
It would make sense to copy the documentation from https://github.com/kubernetes/kubernetes/pull/112200/files, then someone (like @apelisse) who looks at this interface can see how to use it without having to know that some other, similar interface elsewhere explains it.

Reply:
Done, copy+pasted from your PR. thanks

Reply:
I'm not super excited about this API, one needs to keep two handles to shutdown a factory (the stopCh, AND the factory which is error-prone and smelly). I can also call shutdown on something that hasn't been created. I can shutdown but forget to close the channel (it will hang forever?)

Reply:
I don't necesarily disagree, but this is just copying the same change that as was applied to all other informers (https://github.com/kubernetes/kubernetes/pull/112200) to 2 others that were missed. These already shipped in 1.26 as well so I am not sure they can be changed (?).

I don't mind if the API is changed but I do think it needs to be the same across all informers.

Reply:
I agree consistency is key, thanks!

Reply:
Normally we call the method "Run" instead of "Start" and you just wait for it to exit.

I agree with Antoine that we shouldn't call it "Shutdown" when what we mean is "WaitForFinish". I think I didn't realize that the prior PR was extending the API like that :(

I'm not sure how to make it right now :(

Reply:
I guess it does a bit more than just waiting.

I think we should be consistent with other shutdown flows rather than consistent with a confusing one, if we're going to use a consistency argument. I'm sorry for not asking more questions on the prior PR

Reply:
I had called the new function `WaitForShutdown` initially and then there were concerns about that name that led to a renaming :shrug: 

See https://github.com/kubernetes/kubernetes/pull/112200#discussion_r964306374

Reply:
Maybe "RejectNewAndWaitForShutdown" at least captures what it actually does :(

Apologies for not spending more time on that review :(

Reply:
> Normally we call the method "Run" instead of "Start" and you just wait for it to exit.

I agree that the most consistent thing to do here would be to expose a `Run` which blocks. `Start` could be refactored to call `Run` for backwards compatibility.  Is that something people are interested in?

Reply:
Alex, defining a `Run` sounds good to me. It's not clear without digging in if it is good to make one function call the other or not.

We should definitely leave the existing `Start` functions as-is behavior-wise to not break people.

Reply:
Yeah, I think we can define Run to call Start and the now DOA Shutdown, that'd be a good API with minimal change.